tests/unit/restapi/v1/test_tags.py

```diff
@@ -0,0 +1,413 @@
+
+    - A user registers a tag: {name: "tag"}
+    - The user is able to retrieve information about the tag using the
+      tag id.
+    - The returned information matches the information that was provided
+      during registration.
+    """
+    name = "tag"
+
+    tag_response = actions.register_tag(
+        client,
+        name=name,
+        group_id=auth_account["default_group_id"],
+    )
+    tag_expected = tag_response.get_json()
+
+    assert_tag_response_contents_matches_expectations(
+        response=tag_expected,
+        expected_contents={
+            "name": name,
+            "user_id": auth_account["user_id"],
+            "group_id": auth_account["default_group_id"],
+        },
+    )
+
+    assert_retrieving_tag_by_id_works(
+        client, tag_id=tag_expected["tagId"], expected=tag_expected
+    )
+
+
+@pytest.mark.v1
+def test_tag_search_query(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_tags: dict[str, Any],
+) -> None:
+    """Test that tags can be registered and retrieved using the API with search terms.
+
+    This test validates the following sequence of actions:
+
+    - A user registers three tags: "tag1", "tag2", "tag3".
+    - The user is able to retrieve a list of all registered tags with ...
+    - In all cases, the returned information matches the information that was provided
+      during registration.
+    """
+    tag1_expected = registered_tags["tag1"]
+    tag2_expected = registered_tags["tag2"]
+    tag_expected_list = [tag1_expected, tag2_expected]
+
+    assert_retrieving_all_tags_works(
+        client,
+        expected=tag_expected_list,
+        search="name:*ML*",
+    )
+
+
+@pytest.mark.v1
+def test_cannot_register_existing_tag_name(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_tags: dict[str, Any],
+) -> None:
+    """Test that registering a tag with an existing name fails.
+
+    This test validates the following sequence of actions:
+
+    - A user registers a tag named "tag1".
+    - The user attempts to register a second tag with the same name, which fails.
+    """
+    existing_tag = registered_tags["tag1"]
+
+    assert_registering_existing_tag_name_fails(
+        client, name=existing_tag["name"], group_id=existing_tag["group_id"]
+    )
+
+
+@pytest.mark.v1
+def test_rename_tag(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_tags: dict[str, Any],
+) -> None:
+    """Test that a tag can be renamed.
+
+    This test validates the following sequence of actions:
+
+    - A user registers a tag named "old_name".
+    - The user is able to retrieve information about the "old_name" tag that
+      matches the information that was provided during registration.
+    - The user renames this same tag to "new_name".
+    - The user retrieves information about the same tag and it reflects the name
```

Comment:
Documentation: I don't think the tag starts with the name `"old_name"` anymore, from what I can tell it's `"ML: Object Recognition"`.

Reply:
Fixed in commit: [d6b7e43](https://github.com/usnistgov/dioptra/pull/455/commits/d6b7e4340aac661585c95919bcb0095b00d53854)